workflow-core/src/commonMain/kotlin/com/squareup/workflow1/WorkflowIdentifier.kt

```diff
@@ -58,6 +58,21 @@ public class WorkflowIdentifier internal constructor(
 
   private val proxiedIdentifiers = generateSequence(this) { it.proxiedIdentifier }
 
+  private val cachedToString by lazy {
```

Comment:
FYI: Each `by lazy` allocates an additional private object to store the property delegate. It's cheaper to just use a nullable backing property and initialize it yourself. Usually matters more for framework code, so we might want to avoid `by lazy` here.

E.g.
```kotlin
private var _cachedToString: String?
private val cachedToString: String
  get() = _cachedToString ?: calculateToString().also { _cachedToString = it }
```

Reply:
Did that but made cachedToString volatile because I'm not 100% this is actually single threaded. volatile is still better than using synchronization primitives as there's no locking, and the downside is that we might sometimes compute it twice then set it, which is ok.